ProjFS.Mac/PrjFSLib.Mac.Managed/VirtualizationInstance.cs

```diff
@@ -205,7 +206,7 @@ public virtual Result StopVirtualizationInstance()
                     return Result.Success;
 
                 case NotificationType.HardLinkCreated:
-                    this.OnHardLinkCreated(relativePath);
+                    this.OnHardLinkCreated(relativePath, relativeFromPath);
```

Comment:
What would you think of doing the `null` check here?

We'd still have to handle the `null` when logging in the native code, but by doing the check here we could restrict it to `HardLinkCreated`.

Reply:
Thinking about this more, the other calls expect to not have null (and don't check for it).  I think we should ensure the provider doesn't send null across to be consistent.  

If we think we should check for null here, then we should really update all the calls for the possibility of null.

What are your thoughts regarding?

Reply:
Ahh good point.

I think I'm leaning toward making sure the kext just doesn't send null for `relativeFromPath` (for 2-path messages).  What do you think?

Reply:
That's what I got with the latest updates.  Hopefully it's clear.  

For 1 path messages the From can be null, but we shouldn't be using it.  I've been debating if we should send string.empty for non-two path messages, but since those shouldn't use them I'm thinking null for now.